src/Type/Php/ArraySliceFunctionReturnTypeExtension.php

```diff
@@ -50,7 +51,7 @@ public function getTypeFromFunctionCall(FunctionReflection $functionReflection,
 		}
 
 		if ($valueType->isIterableAtLeastOnce()->yes()) {
-			return $valueType->toArray();
+			return TypeCombinator::union($valueType->toArray(), new NonEmptyArrayType());
```

Comment:
intersect, not union

Reply:
I'm trying to make a `non-empty-array&array<T>` into an `array<T>`. Isn't it a union?

Reply:
So you mean that you want an array that’s non-empty to no longer be non-empty? I’m afraid you need to recreate it from scratch by doing new ArrayType.